sway-types/src/span.rs

```diff
@@ -11,59 +11,27 @@ lazy_static! {
     static ref DUMMY_SPAN: Span = Span::new(Arc::from(""), 0, 0, None).unwrap();
 }
 
-pub struct Position {
-    input: Arc<str>,
+pub struct Position<'a> {
+    input: &'a str,
     pos: usize,
 }
 
-impl Position {
-    pub fn new(input: Arc<str>, pos: usize) -> Option<Position> {
-        input.clone().get(pos..).map(|_| Position { input, pos })
+impl<'a> Position<'a> {
+    pub fn new(input: &'a str, pos: usize) -> Option<Position<'a>> {
+        input.get(pos..).map(|_| Position { input, pos })
     }
+
     #[inline]
     pub fn line_col(&self) -> (usize, usize) {
```

Comment:
This function actually outputs different results now, right? Char is shifted back by 1. Should we write a test for this function? I'm surprised a change like that wouldn't break things outside of LSP.

Reply:
You're right good catch. I've changed it back so it outputs the same results as before and then subtracting 1 in the server instead.